thanks/package_tools.py

```diff
@@ -15,31 +19,94 @@ def get_local_dist(package_name):
 
 
 def get_dist_metadata(dist):
-    metadata_path = '{}/{}-{}.dist-info/{}'.format(
-        dist.location,
-        dist.project_name,
-        dist.parsed_version.public,
-        'metadata.json',
-    )
+    metadata_path = get_local_dist_metadata_filepath(dist)
     with open(metadata_path) as fh:
-        metadata = json.load(fh)
+        metadata = parse_metadata(fh.read())
     return metadata
 
 
 def get_funding_data(metadata):
-        return (
-            metadata['extensions']['python.details']['project_urls']['Funding']
-        )
+    print(metadata)
+    return metadata.get('funding_url')
 
 
-def get_local_funding_metadata(package_name):
+def get_local_dist_metadata_filepath(dist):
+    # Dist filename syntax
+    # name ["-" version ["-py" pyver ["-" required_platform]]] "." ext
+    # https://setuptools.readthedocs.io/en/latest/formats.html#filename-embedded-metadata
+    filename = ''.join(chain(*takewhile(lambda x: x[1], (
```

Comment:
This looks cool as hell; could you add a comment explaining what this is doing? These are parts of itertools I'm not familiar with